src/editors/containers/VideoEditor/components/VideoSettingsModal/components/DurationWidget/hooks.js

```diff
@@ -0,0 +1,219 @@
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import { actions, selectors } from '../../../../../../data/redux';
+import messages from '../messages';
+
+import * as module from './hooks';
+
+const durationMatcher = /^(\d{0,2}):?(\d{0,2})?:?(\d{0,2})?$/i;
+
+export const durationWidget = ({ dispatch }) => {
+  const reduxStartStopTimes = useSelector(selectors.video.duration);
+  const setReduxStartStopTimes = (val) => dispatch(actions.video.updateField({ duration: val }));
+  const initialState = module.durationString(reduxStartStopTimes);
```

Comment:
There is no need for this, as the `durationString` function is in the same file. You can just move the function to the top as `const durationString = (duration) => ({` and export it at the bottom. Here, you can call `setUnsavedStartStopTimes(durationString(reduxStartStopTimes` instead.